applications/GeoMechanicsApplication/custom_strategies/builder_and_solvers/residualbased_block_builder_and_solver_linear_elastic_dynamic.h

```diff
@@ -0,0 +1,703 @@
+        const auto base_default_parameters = BaseType::GetDefaultParameters();
+        default_parameters.RecursivelyAddMissingParameters(base_default_parameters);
+        return default_parameters;
+    }
+
+    /**
+     * @brief Returns the name of the class as used in the settings (snake_case format)
+     * @return The name of the class
+     */
+    static std::string Name() { return "block_builder_and_solver_linear_elastic_dynamic"; }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "ResidualBasedBlockBuilderAndSolverLinearElasticDynamic";
+    }
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+private:
+    TSystemMatrixType mMassMatrix;
+    TSystemMatrixType mDampingMatrix;
+    TSystemVectorType mPreviousExternalForceVector;
+    TSystemVectorType mCurrentExternalForceVector;
+
+    TSystemVectorType mPreviousOutOfBalanceVector;
+    TSystemVectorType mCurrentOutOfBalanceVector;
+
+    double mBeta;
+    double mGamma;
+    bool   mCalculateInitialSecondDerivative;
+    bool   mCopyExternalForceVector = false;
+    bool   mUsePerformSolutionStep  = false;
+
+    void BuildRHSNoDirichlet(typename TSchemeType::Pointer pScheme, ModelPart& rModelPart, TSystemVectorType& rb)
+    {
+        // getting the array of the conditions
+        const ConditionsArrayType& r_conditions = rModelPart.Conditions();
+
+        mCurrentExternalForceVector = TSystemVectorType(BaseType::mEquationSystemSize, 0.0);
+
+        // assemble all conditions
+        const auto& r_current_process_info = rModelPart.GetProcessInfo();
+        block_for_each(r_conditions, [&r_current_process_info, this](Condition& r_condition) {
+            LocalSystemVectorType           local_external_force = LocalSystemVectorType(0);
+            Condition::EquationIdVectorType equation_ids;
+
+            if (r_condition.IsActive()) {
+                r_condition.CalculateRightHandSide(local_external_force, r_current_process_info);
+                r_condition.EquationIdVector(equation_ids, r_current_process_info);
+
+                // assemble the elemental contribution
+                BaseType::AssembleRHS(mCurrentExternalForceVector, local_external_force, equation_ids);
+            }
+        });
+
+        // Does: mCurrentOutOfBalanceVector = mCurrentExternalForceVector - mPreviousExternalForceVector;
+        TSparseSpace::ScaleAndAdd(1.0, mCurrentExternalForceVector, -1.0,
+                                  mPreviousExternalForceVector, mCurrentOutOfBalanceVector);
+
+        // Add constraint to the out of balance force before mass and damping components are added, since the mass and damping components are
+        // already constraint by the constraint mass and damping matrix.
+        if (rModelPart.MasterSlaveConstraints().size() != 0) {
```

Comment:
because i have more matrices and vectors storedt, each matrix and vector which interacts which other constrained matrices/ vectors have to be constrained